include/ck/utility/data_type.hpp

```diff
@@ -156,21 +178,1208 @@ struct scalar_type<bf8_t>
     static constexpr index_t vector_size = 1;
 };
 
-template <typename T>
-struct vector_type<T, 1>
+template <>
+struct scalar_type<_BitInt(8)>
 {
-    using d1_t = T;
-    using type = d1_t;
+    using type                           = _BitInt(8);
+    static constexpr index_t vector_size = 1;
+};
+
+int static err = 0;
+
+template <>
+struct vector_type<f8_t, 1>
```

Comment:
Could we template the vector_type instead of reimplementing it for each new data type?